android/src/main/java/com/reactlibrary/QonversionModule.java

```diff
@@ -65,16 +67,31 @@ public String getName() {
     }
 
     @ReactMethod
-    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion) {
-        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(getCurrentActivity().getApplication()).edit();
+    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion,final Promise promise) {
```

Comment:
I think there is no need to resolve and reject the promise here. You can remove `final Promise promise` from the function interface.
Looks like the right way is just to store SDK info if `currencActivity` is null. 
And then try to store it to `SharedPreferences` when `Qonversion.launch()` is called.

Reply:
Done